package/csac_daemon/src/csac_daemon.c

```diff
@@ -0,0 +1,307 @@
+  msg_csac_telemetry_t *msg = alloca(SBP_FRAMING_MAX_PAYLOAD_SIZE);
+  memcpy(msg->telemetry, ctx->csac_telem, telem_string_len);
+  msg->id = 0;
+  msg->telemetry[telem_string_len] = '\0'; /* Ensure null termination */
+  sbp_tx_send(sbp_pubsub_tx_ctx_get(ctx->sbp_ctx),
+              SBP_MSG_CSAC_TELEMETRY,
+              (u8)(0xFF & (telem_string_len + 1)),
+              (u8 *)msg);
+  count++;
+  if ((count % HEADER_SEND_DIVISOR) == 0) {
+    memcpy(msg->telemetry, ctx->csac_header, telem_string_len);
+    telem_string_len = strnlen(ctx->csac_header, 254);
+    msg->telemetry[telem_string_len] = '\0'; /* Ensure null termination */
+    sbp_tx_send(sbp_pubsub_tx_ctx_get(ctx->sbp_ctx),
+                SBP_MSG_CSAC_TELEMETRY_LABELS,
+                (u8)(0xFF & (telem_string_len + 1)),
+                (u8 *)msg);
+  }
+}
+
+
+static int get_csac_telem(struct csac_ctx_s *ctx)
+{
+  static bool headers_received = false;
+  char *buffer = ctx->csac_telem;
+  if (!headers_received) {
+    buffer = ctx->csac_header;
+    for (int tries = 0; tries < 3 && !headers_received; tries++) {
+      memset(buffer, 0, 255);
+      if (write(ctx->port->fd, "6\r\n", 3) < 3) {
+        piksi_log(LOG_DEBUG, "Less than 3 bytes could be written to serial port");
+        continue;
+      }
+      usleep(POLL_SLEEP_MICROSECONDS);
+      int bytes = read_until_cr_lf(ctx->port->fd, buffer, 255);
+      /* Check for expected length and expected first and last chars*/
+      if (bytes > 50) {
+        if (buffer[0] == 'S' && buffer[bytes - 3] == 'r') {
+          headers_received = true;
+        }
+      }
+    } /* end for (int tries ... */
+  }   /* end if !(headers_received) */
+  for (int tries = 0; tries < 3; tries++) {
+    buffer = ctx->csac_telem;
+    /* send query telemetry command followed by a CR LF */
+    if (write(ctx->port->fd, "^\r\n", 3) < 3) {
+      continue;
+    }
+    usleep(POLL_SLEEP_MICROSECONDS);
+    int bytes = read_until_cr_lf(ctx->port->fd, buffer, 255);
+    if (bytes > 50 && headers_received) {
+      return 0;
+    }
+  }
+  return 1; /* either we haven't yet gotten the headers or we didn't get a long
+               enough telemetry string */
+}
+
+/**
+ * @brief csac_telem_callback - used to trigger csac telemetry pollling
+ */
+static void csac_telem_callback(pk_loop_t *loop, void *timer_handle, void *context)
+{
+  (void)loop;
+  (void)timer_handle;
+  struct csac_ctx_s *csac_ctx = (struct csac_ctx_s *)context;
+
+  if (csac_daemon_enabled()) {
+    if (get_csac_telem(csac_ctx) == 0) {
+      send_csac_telem(csac_ctx);
+    } else {
+      piksi_log(LOG_WARNING | LOG_SBP, "Could not retrieve CSAC telemetry. Ensure CSAC is connected and UART0 mode is disabled.");
+    }
+  }
+}
+
+static int settings_changed(void *context)
+{
+  struct csac_ctx_s* csac_ctx = (struct csac_ctx_s *) context;
+  if (!csac_telemetry_enabled) {
+    piksi_log(LOG_WARNING | LOG_SBP, "CSAC telmetry disabled");
```

Comment:
Do we need to leave this enabled?

Reply:
I will make these debug - I agree they don't need to be warnings.